### PR TITLE
Fixes to Rules.mk

### PR DESCRIPTION
**Pull request by bluehood**

### CODE REVIEW

**Comment by bluehood** [comment]
@Axel-Naumann I'm appending rather than prepending to CXXFLAGS so that ROOT's flags take priority -- is that ok?

File: scripts/Rules.mk [hunk 1]
```diff
@@ -322,9 +322,7 @@ ifeq ($(ROOTBITS),)
    export ROOTBITS := $(shell root.exe -b -q -n $(ROOTTEST_HOME)/scripts/Bits.C | grep Bits_in_long | awk '{print $$2;}' )
 endif
 
-ifeq ($(CXXFLAGS),)
-   export CXXFLAGS := $(shell root-config --cflags)
-endif
+export CXXFLAGS := "${CXXFLAGS} $(shell root-config --cflags)"
```
Comment (pcanal) [inline scripts/Rules.mk]: One challenge is that the makefile execution are recursive while drilling down the sub-directories.  This is (likely to be) adding the root flags for each sub-directory.

Comment (bluehood) [reply scripts/Rules.mk]: ah, didn't know that :sweat_smile: 

the problem is that if `CXXFLAGS` is already set in the user environment, as things are we never add `root-config --cflags` to it. Any idea what would be a good fix?

Comment (Axel-Naumann) [reply scripts/Rules.mk]:
```
ifeq($(WE_HAVE_ADDED_ROOT_CFLAGS_TO_CXXFLAGS,))
  export CXXFLAGS := "${CXXFLAGS} $(shell root-config --cflags)"
  WE_HAVE_ADDED_ROOT_CFLAGS_TO_CXXFLAGS:=1
endif
```
?

Comment (bluehood) [reply scripts/Rules.mk]: can do!

Comment (pcanal) [reply scripts/Rules.mk]: Indeed, we already do this a few places we the prefix ROOTTEST_, eg. ROOTTEST_CHECKED_VERSION

**Comment by bluehood** [comment]
@pcanal this seems to fix the original issue (and not cause havoc otherwise), I will merge this afternoon if that's ok

**Pull request merged**